### PR TITLE
Add interactivity to app : Fifth Milestone

### PR DESCRIPTION
**Pull request by moinkhanif**
This pull request features:
- Connecting calculate and operate logic with frontend
- Optimizing calculate and operate modules
- Styling components

### CODE REVIEW

File: src/components/App/App.jsx [hunk 1]
```diff
@@ -1,14 +1,34 @@
+/* eslint-disable react/destructuring-assignment */
```
Comment (Forison) [inline src/components/App/App.jsx]: **[OPTIONAL]** 
By disabling eslint rule, you risk missing out on the object of the rule.

File: src/components/App/App.jsx [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
-// eslint-disable-next-line no-unused-vars
 import calculate from '../logic/calculate';
 
-const App = () => (
-  <div className="main-container">
-    <Display result="0" />
-    <ButtonPanel />
-  </div>
-);
+class App extends Component {
```
Comment (Forison) [inline src/components/App/App.jsx]: **[OPTIONAL]** 
per the convention of container/component pattern, components with their own state ought to be in the container folder, whilst other stateless ones which depends on props in the component folder. After changing App.js to class component it no longer qualifies to be in the component folder.

File: src/components/Button/button.component.jsx [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+  const { name, title } = props;
   return (
-    <button type="button" onClick={handleClick} className="calculator-key">{ name }</button>
+    <button type="button" onClick={handleClick(props)} className="calculator-key" title={title}>{ name }</button>
   );
 };
 
 Button.propTypes = {
   name: propTypes.string,
+  title: propTypes.string,
```
Comment (Forison) [inline src/components/Button/button.component.jsx:15]: **[Required]**
Let us validate handleClick as well. All props have to be validated.

File: src/components/ButtonPanel/button-panel.component.jsx [hunk 4]
```diff
@@ -14,7 +16,22 @@ const ButtonPanel = () => {
...
+          {groups[group]
+            .map((name, j) => {
+              let title;
+              let buttonName = name;
+              if (name.includes(',')) {
+                [buttonName, title] = name.split(',');
+              }
+              return (
+                <Button
+                  clickHandler={handleClick(props)}
```
Comment (Forison) [inline src/components/ButtonPanel/button-panel.component.jsx]: **[Required]**
Let us validate `handleClick` as well. All props have to be validated.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes Required❌
# 🙋‍♂️ @moinkhanif 
#### You have done a good job on this milestone so far, however, there are a few changes we have to make to get the project approved🤗.
* [x] When I divide a number by zero, the application throws an error instead of returning undefined. ie `20/0` will throw an error.
* [x] Let us improve the behavior of percentage thus
- When the percentage is selected after a number it should return the result of the number divided by 100. ie `30` then `%` should display `0.3`
- When percentage is selected at the time number1, number2 and operation are selected, the selection of percentage should return the result of the computation in percentage thus the resultant value over 100. ie `20+40` then `%` should return `0.6`
* [x] Let us improve the behavior of `%`, it ought to mutate each of the two operand not the symbol per se. ie  `20+40` then select `%` should return `-20-40` or `-20+-40`
 - [x] Check other comments below👇
#### That is it kindly make the changes and submit a re-review💪
#### Happy coding🙏
Addo Forison

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes required

Well done @moinkhanif :clap:. You have handled all the requirements from the previous review. Good job adding the styles to your project as well. However, we need to finish the README file before finishing this project. 

Let's add more detailed instructions on how to clone, setup, run this project. Let's also update the status of the fifth milestone by adding a short summary about this milestone.


Please handle this requirement and submit a new code review.

##### Muhammad Ebeid

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @moinkhanif :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**